binary_transparency/firmware/cmd/flash_tool/impl/flash_tool.go

```diff
@@ -157,36 +164,77 @@ func getConsistencyFunc(c *client.ReadonlyClient) func(from, to uint64) ([][]byt
 }
 
 // verifyUpdate checks that an update package is self-consistent and returns a verified proof bundle
-func verifyUpdate(c *client.ReadonlyClient, up api.UpdatePackage, dev devices.Device) (api.ProofBundle, error) {
+func verifyUpdate(c *client.ReadonlyClient, up api.UpdatePackage, dev devices.Device) (api.ProofBundle, api.FirmwareMetadata, error) {
 	var pb api.ProofBundle
+	var fwMeta api.FirmwareMetadata
 
 	// Get the consistency proof for the bundle
 	dc, err := dev.DeviceCheckpoint()
 	if err != nil {
-		return pb, fmt.Errorf("failed to fetch the device checkpoint: %w", err)
+		return pb, fwMeta, fmt.Errorf("failed to fetch the device checkpoint: %w", err)
 	}
 
 	cpFunc := getConsistencyFunc(c)
 	fwHash := sha512.Sum512(up.FirmwareImage)
-	pb, err = verify.BundleForUpdate(up.ProofBundle, fwHash[:], dc, cpFunc)
+	pb, fwMeta, err = verify.BundleForUpdate(up.ProofBundle, fwHash[:], dc, cpFunc)
 	if err != nil {
-		return pb, fmt.Errorf("failed to verify proof bundle: %w", err)
+		return pb, fwMeta, fmt.Errorf("failed to verify proof bundle: %w", err)
 	}
-	return pb, nil
+	return pb, fwMeta, nil
 }
 
-// verifyWitness checks that an update package is consistent with witness
-func verifyWitness(c *client.ReadonlyClient, pb api.ProofBundle, wc client.WitnessClient) error {
+func verifyWitness(c *client.ReadonlyClient, pb api.ProofBundle, witnessURL string) error {
+	wURL, err := url.Parse(witnessURL)
+	if err != nil {
+		return fmt.Errorf("witness_url is invalid: %w", err)
+	}
+	wc := client.WitnessClient{URL: wURL}
+
 	wcp, err := wc.GetWitnessCheckpoint()
 	if err != nil {
 		return fmt.Errorf("failed to fetch the witness checkpoint: %w", err)
 	}
+	if err := verifyRemote(c, pb, *wcp); err != nil {
+		return fmt.Errorf("failed to verify update with witness: %w", err)
```

Comment:
"verify update consistency against" just to be clear that we're not sending the update off to the witness or anything

Reply:
Done.